### PR TITLE
Shopping cart

### PR DESCRIPTION
**Pull request by Tbotny**

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: src/main/java/pages/CheckoutPage.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package pages;
+
+import driver.Driver;
+
+public class CheckoutPage  extends BasePage{
```
Comment (light-di) [inline src/main/java/pages/CheckoutPage.java:5]: Already exist

File: src/main/java/pages/CategoryPage.java [hunk 2]
```diff
@@ -6,9 +6,10 @@
 
 import java.util.List;
 
-public class CategoryPage extends BasePage{
+public class CategoryPage extends BasePage {
     private ProductActions productActions;
     private List<ProductItem> productItemList;
+    private ProductItem productItem;
```
Comment (light-di) [inline src/main/java/pages/CategoryPage.java]: We have productItemList

File: src/main/java/pages/BasePage.java [hunk 3]
```diff
@@ -5,6 +5,7 @@
 import model.Header;
 import model.TopNavBar;
 import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.WebDriverWait;
```
Comment (light-di) [inline src/main/java/pages/BasePage.java]: For what?

File: src/main/java/model/ProductItem.java [hunk 4]
```diff
@@ -46,4 +46,5 @@ public String getProductTax(){
     public String getProductName() {
     	return productNameLink.getText();
     }
+
```
Comment (light-di) [inline src/main/java/model/ProductItem.java]: For what?

File: src/main/java/Main.java [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import org.openqa.selenium.WebDriver;
-
+import pages.HomePage;
```
Comment (light-di) [inline src/main/java/Main.java]: For what?

**Review by TolkachDima — APPROVED** [review]

File: src/main/java/pages/HomePage.java [hunk 6]
```diff
@@ -1,13 +1,25 @@
...
         super(driver);
     }
+    public CategoryMenu getCategoryMenu() {
+        return categoryMenu;
+    }
+    public ProductItem getProductItem() {
+        return productItem;
+    }
 
+    public String getLinkUrl(Driver driver) {
```
Comment (light-di) [inline src/main/java/pages/HomePage.java]: This method should be in step

File: src/main/java/pages/ProductPage.java [hunk 7]
```diff
@@ -24,6 +24,9 @@
     private Link reviews;
     @FindBy(xpath = "//div[@class='alert alert-success alert-dismissible']")
     private TextBlock successMessage;
+    @FindBy(xpath = "//div[@class='alert alert-danger alert-dismissible']")
+    private TextBlock waringMessage;
```
Comment (light-di) [inline src/main/java/pages/ProductPage.java]: Typo, should be "warningMessage"

File: src/main/java/pages/ProductPage.java [hunk 8]
```diff
@@ -33,31 +36,46 @@ public ProductReview getProductReview() {
         return productReview;
     }
 
-    public void goToModelProductReview() {
-        reviews.click();
+    public String getLinkUrl(Driver driver) {
```
Comment (light-di) [inline src/main/java/pages/ProductPage.java]: This method should be in step

File: src/main/java/pages/ProductPage.java [hunk 9]
```diff
@@ -33,31 +36,46 @@ public ProductReview getProductReview() {
...
     }
+
+    public void goToModelProductReview() {
+        reviews.click();
+    }
+
+    public String getSuccessMessage() {
+        return successMessage.getText();
+    }
+    public String getWaringMessage() {
```
Comment (light-di) [inline src/main/java/pages/ProductPage.java]: Typo

File: src/main/java/steps/CheckoutStep.java [hunk 10]
```diff
@@ -0,0 +1,14 @@
+package steps;
+
+import driver.Driver;
+import pages.CheckoutPage;
+
+
+public class CheckoutStep extends BaseStep {
```
Comment (light-di) [inline src/main/java/steps/CheckoutStep.java:7]: Already exist

File: src/main/java/steps/ProductPageStep.java [hunk 11]
```diff
@@ -12,24 +11,85 @@ public ProductPageStep(Driver driver) {
...
+
+    public ProductPageStep addProductToShoppingCartStep() {
         productPage.addProductToShoppingCart();
+        return this;
+    }
+    public ProductPageStep fillFieldProductQuantityStep(String quantity) {
+        productPage.fillFieldProductQuantity(quantity);
+        return this;
+    }
+//        public ProductPageStep fillFieldProductQuantityStep(int quantity) {
```
Comment (light-di) [inline src/main/java/steps/ProductPageStep.java]: Comments are prohibited

File: src/main/java/steps/ProductPageStep.java [hunk 12]
```diff
@@ -12,24 +11,85 @@ public ProductPageStep(Driver driver) {
...
+
+    public ProductPageStep clearFieldProductQuantityStep() {
+        productPage.clearFieldProductQuantity();
+        return this;
+    }
+
+    public String getSuccessMessageAboutAddedProductToShoppingCartStep() {
         return productPage.getSuccessMessage();
     }
+    public String getWaringMessageAboutAddedProductToShoppingCartStep() {
```
Comment (light-di) [inline src/main/java/steps/ProductPageStep.java:67]: Typo

File: src/main/java/Main.java (hunk 5 above)
Comment (Tbotny) [reply src/main/java/Main.java]: удалила лишние импорты

File: src/test/java/TC_DP194_46.java [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    @ParameterizedTest
+    @CsvFileSource(resources = "/invalidQuantity.csv", numLinesToSkip = 1)
+    public void TCDP194_46(String quantity) {
+        ProductPageStep productPageStep = new ProductPageStep(this.driver);
+        String actual = productPageStep.clearFieldProductQuantityStep()
+                .fillFieldProductQuantityStep(quantity)
+                .addProductToShoppingCartStep()
+                .getWaringMessageAboutAddedProductToShoppingCartStep();
```
Comment (light-di) [inline src/test/java/TC_DP194_46.java:26]: Typo

File: src/main/java/model/ProductItem.java (hunk 4 above)
Comment (Tbotny) [reply src/main/java/model/ProductItem.java]: delete gap

File: src/test/java/TC_DP194_13.java [hunk 14]
```diff
@@ -0,0 +1,25 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import steps.ProductPageStep;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+// Validation continue shopping from cart
+public class TC_DP194_13 extends CommonConditionTest {
+    private static final String URL_HOME_PAGE = "http://34.121.117.87/index.php?route=common/home";
```
Comment (light-di) [inline src/test/java/TC_DP194_13.java:10]: Should be in Interface (PagesURL.java)

File: src/test/java/TC_DP194_2.java [hunk 15]
```diff
@@ -0,0 +1,40 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import steps.ProductPageStep;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+//Validation adding item into shopping cart
+public class TC_DP194_2 extends CommonConditionTest {
+    private static final String PRODUCT_PAGE_URL = "http://34.121.117.87/htc-touch-hd";
```
Comment (light-di) [inline src/test/java/TC_DP194_2.java:10]: Should be in Interface (PagesURL.java)

File: src/main/java/pages/BasePage.java (hunk 3 above)
Comment (Tbotny) [reply src/main/java/pages/BasePage.java]: delete import

File: src/test/java/TC_DP194_21.java [hunk 16]
```diff
@@ -0,0 +1,31 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvFileSource;
+import steps.ProductPageStep;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+//Validation changing quantity of items in cart to correct value
+public class TC_DP194_21 extends CommonConditionTest {
+    private static final String PRODUCT_PAGE_URL = "http://34.121.117.87/htc-touch-hd";
```
Comment (light-di) [inline src/test/java/TC_DP194_21.java:11]: Should be in Interface (PagesURL.java)

File: src/test/java/TC_DP194_26.java [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvFileSource;
+import steps.ProductPageStep;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+//Validation the function of add the product to the shopping cart with valid data
+public class TC_DP194_26 extends CommonConditionTest {
+
+    private static final String PRODUCT_PAGE_URL = "http://34.121.117.87/htc-touch-hd";
```
Comment (light-di) [inline src/test/java/TC_DP194_26.java:12]: Should be in Interface (PagesURL.java)

File: src/test/java/TC_DP194_31.java [hunk 18]
```diff
@@ -0,0 +1,27 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvFileSource;
+import steps.ProductPageStep;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+//Validation the function adding a product review with valid data
+public class TC_DP194_31 extends CommonConditionTest {
+    private static final String PRODUCT_PAGE_URL = "http://34.121.117.87/htc-touch-hd";
```
Comment (light-di) [inline src/test/java/TC_DP194_31.java:11]: Should be in Interface (PagesURL.java)

File: src/test/java/TC_DP194_4.java [hunk 19]
```diff
@@ -0,0 +1,28 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import steps.ProductPageStep;
+import steps.ShoppingCartStep;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+//Validation removing items from cart
+public class TC_DP194_4 extends CommonConditionTest {
+    private static final String URL_HOME_PAGE = "http://34.121.117.87/index.php?route=common/home";
```
Comment (light-di) [inline src/test/java/TC_DP194_4.java:11]: Should be in Interface (PagesURL.java)

File: src/test/java/TC_DP194_46.java [hunk 20]
```diff
@@ -0,0 +1,29 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvFileSource;
+import steps.ProductPageStep;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TC_DP194_46 extends CommonConditionTest {
+
+    private static final String PRODUCT_PAGE_URL = "http://34.121.117.87/htc-touch-hd";
```
Comment (light-di) [inline src/test/java/TC_DP194_46.java:11]: Should be in Interface (PagesURL.java)

File: src/test/java/TC_DP194_8.java [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvFileSource;
+import steps.ProductPageStep;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+//Validation applying correct coupon code
+
+public class TC_DP194_8 extends CommonConditionTest {
+    private static final String SUCCESS_MESSAGE = "Success: Your coupon discount has been applied!\n×";
```
Comment (light-di) [inline src/test/java/TC_DP194_8.java:12]: Should be in Interface (PagesURL.java)

File: src/test/resources/cartQuantity.csv [hunk 22]
```diff
@@ -0,0 +1,4 @@
+quantity
```
Comment (light-di) [inline src/test/resources/cartQuantity.csv:1]: Add pageName to csv

File: src/test/resources/invalidQuantity.csv [hunk 23]
```diff
@@ -0,0 +1,5 @@
+quantity
```
Comment (light-di) [inline src/test/resources/invalidQuantity.csv:1]: Add pageName to csv

File: src/test/resources/validCoupon.csv [hunk 24]
```diff
@@ -0,0 +1,4 @@
+coupon
```
Comment (light-di) [inline src/test/resources/validCoupon.csv:1]: Add pageName to csv

File: src/test/resources/validQuantity.csv [hunk 25]
```diff
@@ -0,0 +1,4 @@
+quantity
```
Comment (light-di) [inline src/test/resources/validQuantity.csv:1]: Add pageName to csv

File: src/test/resources/validReview.csv [hunk 26]
```diff
@@ -0,0 +1,4 @@
+name,review,message
```
Comment (light-di) [inline src/test/resources/validReview.csv:1]: Add pageName to csv

File: src/main/java/pages/CategoryPage.java (hunk 2 above)
Comment (Tbotny) [reply src/main/java/pages/CategoryPage.java]: delete  private ProductItem productItem

File: src/main/java/pages/CheckoutPage.java (hunk 1 above)
Comment (Tbotny) [reply src/main/java/pages/CheckoutPage.java:5]: need gag

File: src/main/java/pages/ProductPage.java (hunk 7 above)
Comment (Tbotny) [reply src/main/java/pages/ProductPage.java]: warning

Comment (Tbotny) [reply src/main/java/pages/ProductPage.java]: fixed

**Review by ghost — APPROVED** [review]

**Review by light-di — APPROVED** [review]

**Pull request merged**